### ISSUE TITLE
reflect.Value.Convert: value of type []struct { Name string; Children []xreflect.Forward } cannot be converted to type []xreflect.Forward

### ISSUE THREAD

**Issue by RafalSkolasinski** [gopherdata/gophernotes#208]
When trying to create a simple graph structure - a `struct` containing a slice of itself - I am getting problems in `gophernotes`. 

Same works fine in just plain `go` file.

Code:
```
package main

import "fmt"

type Item struct {
    Name               string
    Children           []Item
}

graph := Item{
    Name: "my-name",
    Children: []Item{
        {Name: "other-name"},
    },
}
```

throws error (warning?)
```
reflect.Value.Convert: value of type []struct { Name string; Children []xreflect.Forward } cannot be converted to type []xreflect.Forward
```

**Comment by cosmos72** [comment]
There are known limitations on recursive types, because they are emulated: the `reflect` package cannot create them.
Yet, this should be one of the supported cases... thanks for pointing out, I will investigate

**Comment by RafalSkolasinski** [comment]
Would consider adding some warning in the readme for a meantime. It was very confusing.

**Comment by cosmos72** [comment]
README.md updated

**Comment by cosmos72** [comment]
Fixed in commit 39e156a960494a8c772801bf5d066e85800d3f6a
